node/src/components/linear_chain_sync/operations.rs

```diff
@@ -463,12 +462,9 @@ pub(crate) async fn run_fast_sync_task(
                 .take_header();
     }
 
-    // The era supervisor needs validator information from previous eras.
-    // The number of previous eras is determined by a *delay* in which consensus participants become
-    // bonded validators or unbond.
-    let delay = consensus::bonded_eras(&(&chainspec).into());
-    // The era supervisor requires at least to 3*delay + 1 eras back to be stored in the database.
-    let historical_eras_needed = delay.saturating_mul(3).saturating_add(1);
+    // The era supervisor requires at least auction_delay + 3 eras back to be stored in the
+    // database.
+    let historical_eras_needed = chainspec.core_config.auction_delay.saturating_add(3);
```

Comment:
Could we hide it behind `consensus::bonded_eras` (or similar) as well? I'm not a fan of hardcoding these assumptions all over the code. It's easy to miss and get out of sync.

Reply:
I'm working on it. But I'll have to change the `EraSupervisor` to keep a `Chainspec` instead of a `ProtocolConfig`… that type should probably be removed anyway at some point.

Reply:
Done in https://github.com/casper-network/casper-node/pull/2127/commits/429a2b1af6acf011c0a141bc198e9bb52ecb75b8.